lib/KeyManager.php

```diff
@@ -712,4 +703,23 @@ public function getMasterKeyId() {
 	public function getPublicMasterKey() {
 		return $this->keyStorage->getSystemUserKey($this->masterKeyId . '.publicKey', Encryption::ID);
 	}
+
+	/**
+	 * set publicShareKeyId and masterKeyId if not set
+	 */
+	public function setPublicShareKeyIDAndMasterKeyId() {
+		$this->publicShareKeyId = $this->config->getAppValue('encryption',
+			'publicShareKeyId');
+		if (empty($this->publicShareKeyId)) {
+			$this->publicShareKeyId = 'pubShare_' . substr(md5(time()), 0, 8);
+			$this->config->setAppValue('encryption', 'publicShareKeyId', $this->publicShareKeyId);
+		}
+
+		$this->masterKeyId = $this->config->getAppValue('encryption',
+			'masterKeyId');
+		if (empty($this->masterKeyId)) {
```

Comment:
don't use `empty()` for strings, because in PHP `empty("0")` is true...

Reply:
same above

Reply:
Instead of using `empty()` , tried to use `!== ''`

Reply:
can it be null ? in your other PRs you tested with both `is_null` and `!== ''`

Reply:
Updated the check for `is_null` and `!== ''`